include/Gaffer/ShufflePlug.inl

```diff
@@ -77,7 +77,7 @@ T ShufflesPlug::shuffle( const T &sourceContainer ) const
 
 	NameContainer names;
 
-	for( const ConstShufflePlugPtr &plug : ShufflePlug::Range( *this ) )
+	for( auto &plug : ShufflePlug::Range( *this ) )
```

Comment:
My personal preference would be `const ShufflePlugPtr &` rather than `auto &` so that I can see the type of plug, rather than having to go read the declaration of ShufflePlug::Range() and work out the type that will be deduced by the compiler. Personallly I think `auto` makes sense when the deduced type is some complex meta-programming type, but for general code I think it makes it harder to reason about the code as the reader has to do the type deduction themselves rather than just having the type explicitely there in front of them.